clients/fides-js/src/lib/consent.tsx

```diff
@@ -112,6 +126,49 @@ const getLocation = async (
   }
 };
 
+/**
+ * Hide pre-existing link in the DOM if we do not need to trigger a modal
+ */
+const hideModalLink = (debug: boolean): void => {
+  // TODO- it's possible that this element does not exist by the time this method runs
+  const modalLinkEl: HTMLElement | null =
+    document.getElementById(FIDES_MODAL_LINK);
+  if (modalLinkEl) {
+    debugLog(debug, "modal link element exists, attempting to hide it");
+    // TODO: hide link
+    // eslint-disable-next-line no-param-reassign
+    modalLinkEl.style.display = "none";
+  }
+  debugLog(
+    debug,
+    "modal link element does not exist, so there is nothing to hide"
+  );
+};
+
+/**
+ * Update the pre-existing modal link in the DOM to trigger the modal
+ */
+const bindModalLinkToModal = (debug: boolean): void => {
+  // TODO- it's possible that this element does not exist by the time this method runs
+  const modalLinkEl: HTMLElement | null =
+    document.getElementById(FIDES_MODAL_LINK);
+  if (
+    modalLinkEl &&
+    // TODO: does this need to be an HTMLAnchorElement?
+    modalLinkEl instanceof HTMLAnchorElement
+  ) {
+    debugLog(
+      debug,
+      `Fides modal link element found, updating click event to trigger modal`
+    );
+    modalLinkEl.onclick = () => {
+      // TODO: render modal component
```

Comment:
hmm this might be tricky, I'll have to think about how we can trigger it from here...